fprettify/__init__.py

```diff
@@ -228,6 +228,180 @@
+F90_PROCEDURES_RE = re.compile(r"\b(" + "|".join((
+    "abs", "achar", "acos", "adjustl", "adjustr", "aimag", "aint",
+    "all", "allocated", "anint", "any", "asin", "associated",
+    "atan", "atan2", "bit_size", "btest", "ceiling", "char", "cmplx",
+    "conjg", "cos", "cosh", "count", "cshift", "date_and_time", "dble",
+    "digits", "dim", "dot_product", "dprod", "eoshift", "epsilon",
+    "exp", "exponent", "floor", "fraction", "huge", "iachar", "iand",
+    "ibclr", "ibits", "ibset", "ichar", "ieor", "index", "int", "ior",
+    "ishft", "ishftc", "kind", "lbound", "len", "len_trim", "lge", "lgt",
+    "lle", "llt", "log", "log10", "logical", "matmul", "max",
+    "maxexponent", "maxloc", "maxval", "merge", "min", "minexponent",
+    "minloc", "minval", "mod", "modulo", "mvbits", "nearest", "nint",
+    "not", "pack", "precision", "present", "product", "radix",
+    ## Real is taken out here to avoid highlighting declarations.
+    "random_number", "random_seed", "range", ## "real"
+    "repeat", "reshape", "rrspacing", "scale", "scan",
+    "selected_int_kind", "selected_real_kind", "set_exponent",
+    "shape", "sign", "sin", "sinh", "size", "spacing", "spread", "sqrt",
+    "sum", "system_clock", "tan", "tanh", "tiny", "transfer",
+    "transpose", "trim", "ubound", "unpack", "verify",
+    ## F95 intrinsic functions.
+    "null", "cpu_time",
+    ## F2003.
+    "move_alloc", "command_argument_count", "get_command",
+    "get_command_argument", "get_environment_variable",
+    "selected_char_kind", "wait", "flush", "new_line",
+    "extends", "extends_type_of", "same_type_as", "bind",
+    ## F2003 ieee_arithmetic intrinsic module.
+    "ieee_support_underflow_control", "ieee_get_underflow_mode",
+    "ieee_set_underflow_mode",
+    ## F2003 iso_c_binding intrinsic module.
+    "c_loc", "c_funloc", "c_associated", "c_f_pointer",
+    "c_f_procpointer",
+    ## F2008.
+    "bge", "bgt", "ble", "blt", "dshiftl", "dshiftr", "leadz", "popcnt",
+    "poppar", "trailz", "maskl", "maskr", "shifta", "shiftl", "shiftr",
+    "merge_bits", "iall", "iany", "iparity", "storage_size",
+    "bessel_j0", "bessel_j1", "bessel_jn",
+    "bessel_y0", "bessel_y1", "bessel_yn",
+    "erf", "erfc", "erfc_scaled", "gamma", "hypot", "log_gamma",
+    "norm2", "parity", "findloc", "is_contiguous",
+    "sync images", "lock", "unlock", "image_index",
+    "lcobound", "ucobound", "num_images", "this_image",
+    ## F2008 iso_fortran_env module.
+    "compiler_options", "compiler_version",
+    ## F2008 iso_c_binding module.
+    "c_sizeof"
+    )) + r")\b", RE_FLAGS)
+
+## Regexp matching intrinsic operators
+F90_OPERATORS_RE = re.compile(r"(" + "|".join([r"\." + a + r"\." for a in (
+    "and", "eq", "eqv", "false", "ge", "gt", "le", "lt", "ne",
+    "neqv", "not", "or", "true"
+    )]) + r")", RE_FLAGS)
+
+## Regexp for all HPF keywords, procedures and directives.
+F90_HPF_KEYWORDS_RE = re.compile(r"\b(" + "|".join((
```

Comment:
Is there any particular reason for including HPF keywords? Otherwise I'd suggest to restrict keywords to those specified in the Fortran standard.